make/config.mk

```diff
@@ -54,12 +54,29 @@ USE_CUDA_PATH = NONE
 # whether use CuDNN R3 library
 USE_CUDNN = 0
 
-# CUDA architecture setting: going with all of them.
-# For CUDA < 6.0, comment the *_50 lines for compatibility.
-CUDA_ARCH := -gencode arch=compute_30,code=sm_30 \
-		-gencode arch=compute_35,code=sm_35 \
-		-gencode arch=compute_50,code=sm_50 \
-		-gencode arch=compute_50,code=compute_50
+# Sets 'CUDA_ARCH', which determines the GPU architectures supported
+# by the compiled kernels.  Users can edit the KNOWN_CUDA_ARCHS list below
+# to remove archs they don't wish to support to speed compilation.
+#
+# For archs in this list, nvcc will create a fat-binary that will include
+# the binaries (SASS) for all architectures supported by the installed version
+# of the cuda toolkit, plus the assembly (PTX) for the most recent such architecture.
+# If these kernels are then run on a newer-architecture GPU, then the binary will
+# be JIT-compiled by the updated driver from the included PTX.
+#
+# If CUDA is not present on the system, CUDA_ARCH will be quietly set to "".
+KNOWN_CUDA_ARCHS := 30 35 50 52 60 61
+# Run nvcc on a zero-length file to check architecture-level support.
+# Build nvcc arg list for supported levels to include SASS in the fat-binary.
+CUDA_ARCH := $(foreach arch,$(KNOWN_CUDA_ARCHS), \
+                  $(shell nvcc -arch=sm_$(arch) -E --x cu /dev/null >/dev/null 2>&1 && \
```

Comment:
leave `CUDA_ARCH` as empty in `config.mk`. In makefile, if this variable is empty, then assign it with values by using `$(NVCC)`